spec/operators/takeWhile-spec.ts

```diff
@@ -364,4 +364,130 @@ describe('takeWhile', () => {
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const subject = new Subject();
+      const source = cold('-------a----b----c---|');
+      const expected = '   -------a----(b|)';
+      const subs = '       ^-----------!';
+
+      const result = merge(source, subject).pipe(
+        takeWhile((x) => x !== 'b', true),
+        tap({
+          complete: () => {
+            subject.error(new Error('reentrant shennanigans'));
+          },
+        })
+      );
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(source.subscriptions).toBe(subs);
+    });
+  });
+
+  it('should not emit errors sent from the source *after* it found the first value in reentrant scenarios in next', () => {
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const subject = new Subject();
+      const source = cold('-------a----b----c---|');
+      const expected = '   -------a----(b|)';
+      const subs = '       ^-----------!';
+
+      const result = merge(source, subject).pipe(
+        takeWhile((x) => x !== 'b', true),
+        tap((x) => {
+          if (x === 'b') {
+            subject.error(new Error('reentrant shennanigans'));
+          }
+        })
+      );
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(source.subscriptions).toBe(subs);
+    });
+  });
+
+  it('should not emit completes sent from the source *after* it found the first value in reentrant scenarios', () => {
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const subject = new Subject();
+      const source = cold('-------a----b----c---|');
+      const expected = '   -------a----(b|)';
+      const subs = '       ^-----------!';
+
+      const result = merge(source, subject).pipe(
+        takeWhile((x) => x !== 'b', true),
+        tap((x) => {
+          if (x === 'b') {
+            subject.complete();
+          }
+        })
+      );
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(source.subscriptions).toBe(subs);
+    });
+  });
+
+  it('should not emit errors sent from the source *after* it found the first value in reentrant scenarios', () => {
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const subject = new Subject();
+      const source = cold('-------a----b----c---|');
+      const expected = '   -------a----|';
+      const subs = '       ^-----------!';
+
+      const result = merge(source, subject).pipe(
+        takeWhile((x) => x !== 'b'),
+        tap({
+          complete: () => {
+            subject.error(new Error('reentrant shennanigans'));
+          },
+        })
+      );
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(source.subscriptions).toBe(subs);
+    });
+  });
+
+  it('should not emit completions sent from the source *after* it found the first value in reentrant scenarios', () => {
```

Comment:
does not fail on master, had to use
```ts
        tap((x) => {
          if (x === 'b') {
            subject.complete();
          }
        })
```

Reply:
This would not be testing the same thing, exactly.

Ultimately, this and the test below it are really just exercising the subscriber guarantees. But nonetheless I think we should keep them so someone doesn't implement something in the future that breaks this contract.